pkgs/development/libraries/libjpeg-turbo/default.nix

```diff
@@ -1,6 +1,12 @@
-{ lib, stdenv, fetchFromGitHub, cmake, nasm
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, nasm
 , openjdk
 , enableJava ? false # whether to build the java wrapper
+, enableJpeg7 ? false # whether to build libjpeg.so.7
+, enableJpeg8 ? false # whether to build libjpeg.so.8
```

Comment:
Is it allowed to set both of these to `true`? What happens in that case?

If it's not allowed, it would make sense to add an evaluation-time assertion that ensures that only one of them is true.

Also, does `whether to build libjpeg.so.7` mean it just creates the additional file, or is the unversioned `libturbojpeg.so` file then linked to thiis version?

Reply:
I just checked it; looks like the higher version "wins" for `enableJpeg7 = true; enableJpeg8 = true;`:

```
$ ls -lah $(nix-build --no-link --expr '((import ./. {}).libjpeg_turbo.override { enableJpeg7 = true; enableJpeg8 = true; }).out')/lib
total 617K
dr-xr-xr-x 2 root root    8 Jan  1  1970 .
dr-xr-xr-x 3 root root    3 Jan  1  1970 ..
lrwxrwxrwx 1 root root   12 Jan  1  1970 libjpeg.so -> libjpeg.so.8
lrwxrwxrwx 1 root root   16 Jan  1  1970 libjpeg.so.8 -> libjpeg.so.8.2.2
-r-xr-xr-x 1 root root 727K Jan  1  1970 libjpeg.so.8.2.2
lrwxrwxrwx 1 root root   17 Jan  1  1970 libturbojpeg.so -> libturbojpeg.so.0
lrwxrwxrwx 1 root root   21 Jan  1  1970 libturbojpeg.so.0 -> libturbojpeg.so.0.2.0
-r-xr-xr-x 1 root root 797K Jan  1  1970 libturbojpeg.so.0.2.0
```

Please add that to the comments, looks good otherwise.

Reply:
Thanks, I'm gonna check